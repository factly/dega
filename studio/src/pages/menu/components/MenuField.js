import React, { useEffect } from 'react';
import { Form, Input, Collapse } from 'antd';

function MenuField({ field }) {
  const { Panel } = Collapse;
  const [panelHeader, setPanelHeader] = React.useState('Menu');
  const [inputRef, setInputRef] = React.useState(null);
  const handleInputChange = (e) => {
    var header = e.target.value;
    setPanelHeader(header);
  };
  useEffect(() => {
    if (inputRef) {
      setPanelHeader(inputRef.props.value);
    }
  }, [inputRef]);
  return (
    <Collapse
      style={{ maxWidth: '375px', background: '#f0f2f5', border: 0, marginLeft: '25px' }}
      defaultActiveKey={['1']}
    >
      <Panel header={panelHeader} key={['1']}>
        <Form.Item
          {...field}
          fieldKey={[field.fieldKey, 'name']}
          name={[field.name, 'name']}
          label="Navigation Label"
          rules={[
            {
              required: true,
              message: 'Please input the name!',
            },
          ]}
        >
          <Input
            ref={(input) => {
              if (input?.props) setInputRef(input);
            }}
            placeholder="Enter Label"
            onChange={(e) => handleInputChange(e)}
          />
        </Form.Item>
        <Form.Item
          {...field}
          fieldKey={[field.fieldKey, 'title']}
          name={[field.name, 'title']}
          label="Title Attribute"
        >
          <Input placeholder="Enter Title" />
        </Form.Item>
        <Form.Item
          {...field}
          fieldKey={[field.fieldKey, 'url']}
          name={[field.name, 'url']}
          label="URL"
        >
          <Input placeholder="Enter URL" />
        </Form.Item>
      </Panel>
    </Collapse>
  );
}

export default MenuField;
