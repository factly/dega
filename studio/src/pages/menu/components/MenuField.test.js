import React from 'react';
import { mount, shallow } from 'enzyme';
import { act } from '@testing-library/react';
import { Form } from 'antd';

import '../../../matchMedia.mock';
import MenuField from './MenuField';

describe('MenuField component', () => {
  describe('snapshot testing', () => {
    it('should render the component', () => {
      let component;
      act(() => {
        component = shallow(
          <Form>
            <MenuField field={[]} />
          </Form>,
        );
      });
      expect(component).toMatchSnapshot();
    });
  });
  describe('component testing', () => {
    it('should handle input name change', () => {
      let component;
      const field = {
        fieldKey: '1',
        name: '1',
      };
      act(() => {
        component = mount(
          <Form>
            <MenuField field={field} />
          </Form>,
        );
      });
      component
        .find('input')
        .at(0)
        .simulate('change', { target: { value: 'New name' } });
      expect(component.find('CollapsePanel').at(1).props().header).toBe('New name');
      component.unmount();
    });
  });
});
