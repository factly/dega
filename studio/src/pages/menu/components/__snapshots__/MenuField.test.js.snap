// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuField component snapshot testing should render the component 1`] = `
<SizeContextProvider>
  <ContextProvider
    value={
      Object {
        "colon": undefined,
        "itemRef": [Function],
        "labelAlign": undefined,
        "labelCol": undefined,
        "name": undefined,
        "requiredMark": true,
        "vertical": false,
        "wrapperCol": undefined,
      }
    }
  >
    <ForwardRef(Form)
      className="ant-form ant-form-horizontal"
      form={
        Object {
          "__INTERNAL__": Object {
            "itemRef": [Function],
            "name": undefined,
          },
          "getFieldError": [Function],
          "getFieldInstance": [Function],
          "getFieldValue": [Function],
          "getFieldsError": [Function],
          "getFieldsValue": [Function],
          "getInternalHooks": [Function],
          "isFieldTouched": [Function],
          "isFieldValidating": [Function],
          "isFieldsTouched": [Function],
          "isFieldsValidating": [Function],
          "resetFields": [Function],
          "scrollToField": [Function],
          "setFields": [Function],
          "setFieldsValue": [Function],
          "submit": [Function],
          "validateFields": [Function],
        }
      }
      onFinishFailed={[Function]}
    >
      <MenuField
        field={Array []}
      />
    </ForwardRef(Form)>
  </ContextProvider>
</SizeContextProvider>
`;
